src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -324,7 +324,15 @@ func (l *commitLog) Open() error {
 	}
 
 	l.commitLogFailFn = func(err error) {
-		l.log.Fatal("fatal commit log error", zap.Error(err))
+		isFatal := l.opts.FailureStrategy() == FailureStrategyPanic
+		if l.opts.FailureStrategy() == FailureStrategyCallback && !l.opts.FailureCallback()(err) {
+			isFatal = true
+		}
+		if isFatal {
+			l.log.Fatal("fatal commit log error", zap.Error(err))
+		} else {
+			l.log.Error("commit log error", zap.Error(err))
+		}
```

Comment:
nit: Only call FailureStrategy() once here to avoid two fn calls and indirections?
```suggestion
		strategy := l.opts.FailureStrategy()
		fatal := strategy == FailureStrategyPanic
		if strategy == FailureStrategyCallback && !l.opts.FailureCallback()(err) {
			fatal = true
		}
		if fatal {
			l.log.Fatal("fatal commit log write error", zap.Error(err))
		} else {
			l.log.Error("commit log write error", zap.Error(err))
		}
```